pkg/service/service.go

```diff
@@ -847,7 +899,7 @@ func (s *Service) upsertServiceIntoLBMaps(svc *svcInfo, onlyLocalBackends bool,
 		return err
 	}
 
-	if option.Config.EnableSessionAffinity {
+	if option.Config.EnableSessionAffinity && svc.l7LBProxyPort == 0 {
```

Comment:
Please add `(s *svc) IsL7Service()` and call it here.

Reply:
Done :-)